### PR TITLE
zsh: add autocd option

### PR DESCRIPTION
**Pull request by kaliumxyz**
autocd option was missing. It allows one to type a path (relative or non-relative) into their terminal raw and navigate to it. This feature is in the zsh configuration wizard and is why I personally started using ZSH so I hope it can be included c:

### CODE REVIEW

File: modules/programs/zsh.nix [hunk 1]
```diff
@@ -149,6 +149,14 @@ in
     programs.zsh = {
       enable = mkEnableOption "Z shell (Zsh)";
 
+      autocd = mkOption {
+        default = false;
+        description = ''
+        automatically enter into a directory if typed directly into shell.
```
Comment (rycee) [inline modules/programs/zsh.nix]: The description should start with a capital letter, also can fix the indentation:
```suggestion
          Automatically enter into a directory if typed directly into shell.
```

**Comment by rycee** [comment]
Thanks for the contribution. Please have a look at the comment I added.

@uvNikita Mind having a look at this?

**Comment by uvNikita** [comment]
It will definitely be good to include this option, thanks for creating a PR @kaliumxyz!

One issue I noticed is that it seems oh-my-zsh sets this option as well: https://github.com/robbyrussell/oh-my-zsh/blob/76c7bd57c332b3d9231dc2c6167f3c5d3e56e110/lib/theme-and-appearance.zsh#L42. Which means, that after merging these changes, autocd will be disabled for everyone using zsh.

I guess the best way to fix it would be to turn option type to `nullOr bool` with the default value of null and just not set the option by default. We should probably do the same with other options, but this can be done in other PR.

**Comment by kaliumxyz** [comment]
I've included the @rycee and @uvNikita posted.

**Comment by kaliumxyz** [comment]
correction, I have now included it any way to squash these commits?

**Comment by kaliumxyz** [comment]
correction, I need to do more local testing before this can be merged my apologies.

**Comment by kaliumxyz** [comment]
Forgot to actually implement the chance beyond the option, its now corrected. Can I squash the commits somehow or should I open a new PR with a cleaner commit history?

**Comment by rycee** [comment]
You can squash locally and force push the branch. The PR will be updated automatically.

On May 15, 2019 2:12:54 AM GMT+02:00, kaliumxyz <notifications@github.com> wrote:
>Forgot to actually implement the chance beyond the option, its now
>corrected. Can I squash the commits somehow or should I open a new PR
>with a cleaner commit history?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/rycee/home-manager/pull/707#issuecomment-492456015

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by kaliumxyz** [comment]
Okay, I've squashed the last few commits into one cleaner one with a proper description. Thanks for taking the time to explain this to me. Please review and accept the pr c:

**Comment by uvNikita** [comment]
I squashed the rest of the commits to have cleaner history and merged into master: https://github.com/rycee/home-manager/commit/02a07f19a14920597732647932bf5963ed2f1de8. Thanks for your contribution!

**Pull request closed without merging**